### PR TITLE
userFavorites

### PR DESCRIPTION
**Pull request by yjwangit**
Please review. Thank you.

### CODE REVIEW

File: server/db.mjs [hunk 1]
```diff
@@ -3,15 +3,81 @@ import pgp from "pg-promise";
...
     ])
   )[0];
+export const testSearch = async () => {
+  let data = await db.any("SELECT * FROM PARK WHERE PARK_ID IN ('1','2')");
+  console.log(data, "testSearch");
+};
+// testSearch();
+//search usersave parks
+export const getUserFavorites = async (userId) => {
+  const user = await db.any(`SELECT * FROM USERS WHERE USER_ID = '${userId}'`);
```
Comment (sunnybrie) [inline server/db.mjs:21]:
```suggestion
  const user = await db.any(`SELECT * FROM USERS WHERE USER_ID = ‘$1’`, [userId]);
```
Using this format for your database queries might make things easier on you in the future than using template string literals.

**Review by sunnybrie — APPROVED** [review]
Left a suggestion in another comment, but overall this looks fine to me!

File: server/db.mjs (hunk 1 above)
Comment (yjwangit) [reply server/db.mjs:21]: Thank you Brie for your suggestion!

File: app/src/App.js [hunk 2]
```diff
@@ -3,13 +3,11 @@ import React, { useState } from "react";
 import { Route, useHistory } from "react-router-dom";
 
 import Header from "./components/Header";
-import LogoutButton from "./components/Logout";
+import MyFavorites from "./components/MyFavorites";
```
Comment (alexcheva) [inline app/src/App.js:6]: 👍

File: app/src/components/MyFavorites.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+import { useAuth0 } from "@auth0/auth0-react";
+
+const MyFavorites = () => {
+  const { loginWithRedirect } = useAuth0();
```
Comment (alexcheva) [inline app/src/components/MyFavorites.js:6]: nice!

File: app/src/components/Park.js [hunk 4]
```diff
@@ -1,9 +1,31 @@
...
+  const { user, isAuthenticated } = useAuth0();
   // <img src={park.images[0] || ""} />;
-  console.log(park);
+  console.log(park, user);
   const safePark = park || {};
+  const addFavorite = (e) => {
+    e.stopPropagation();
+    if (!isAuthenticated) {
+      alert("please login");
+      return;
```
Comment (alexcheva) [inline app/src/components/Park.js:14]: Great UI!

File: app/src/components/Park.js [hunk 5]
```diff
@@ -1,9 +1,31 @@
...
+      parkName: park.fullName,
+      parkCover: park.images.length > 0 ? park.images[0].url : "",
+      userId: user.sub,
+    };
+    axios
+      .post("http://localhost:4000/api/tasks/addFavorite", params)
+      .then((res) => {
+        console.log(res);
+      });
+  };
```
Comment (alexcheva) [inline app/src/components/Park.js:28]: Nice use of params! Need to do that too ❤️

File: server/server.mjs [hunk 6]
```diff
@@ -33,16 +34,23 @@ tasks.get("/parks", async (request, response) => {
...
 tasks.use(express.json());
-tasks.post("/", async (request, response) => {
-  const { name } = request.body;
-  const task = await db.addTask(name);
-  response.status(201).json(task);
+tasks.post("/addFavorite", async (request, response) => {
+  const { parkId, parkName, parkCover, userId } = request.body;
+  const params = { parkId, parkName, parkCover, userId };
+  await db.insertPark(params);
+  await db.insertUser(params);
```
Comment (alexcheva) [inline server/server.mjs:42]: Nice use of destructuring!

File: server/server.mjs [hunk 7]
```diff
@@ -1,7 +1,7 @@
 import axios from "axios";
 import cors from "cors";
 import dotenv from "dotenv";
-import express from "express";
+import express, { query } from "express";
```
Comment (alexcheva) [inline server/server.mjs:4]: Where do you use query? 🤔  I think maybe it should be - import  { express, query } from "express";

File: server/server.mjs [hunk 8]
```diff
@@ -33,16 +34,23 @@ tasks.get("/parks", async (request, response) => {
...
+  await db.insertUser(params);
+  response.status(201).json({
+    message: "success",
+    code: 201,
+  });
 });
-
-app.use("/api/tasks", tasks);
+tasks.get("/getUserFavorites", async (request, response) => {
+  const { userId } = request.query;
```
Comment (alexcheva) [inline server/server.mjs:49]: Oh, this is what the query for?

**Review by alexcheva — APPROVED** [review]
Looking good!

File: server/server.mjs (hunk 7 above)
Comment (yjwangit) [reply server/server.mjs:4]: It was automatically imported by VScode and has been deleted. Thanks much for your review!

File: server/server.mjs (hunk 8 above)
Comment (yjwangit) [reply server/server.mjs:49]: It's a property of request object.

**Pull request merged**